core/org.wso2.carbon.tomcat/src/main/java/org/wso2/carbon/tomcat/internal/CarbonTomcat.java

```diff
@@ -408,8 +411,18 @@ public void startConnectors(int portOffset) {
         Connector[] connectors = this.getService().findConnectors();
         for (Connector connector : connectors) {
             try {
-                int currentPort = connector.getPort();
-                connector.setPort(currentPort + portOffset);
+                String isRandomPort = System.getProperty("tomcat.random.port.enable");
```

Comment:
Why is this needed? If someone specifies HTTP or HTTPS port system props, then we don't use random port. So this property is not needed.